pip_api/_parse_requirements.py

```diff
@@ -214,6 +219,8 @@ def parse_requirements(
             if req:
                 if not isinstance(req, UnparsedRequirement):
                     req.comes_from = "-r {} (line {})".format(filename, lineno)  # type: ignore
+                    if req.marker is not None and not req.marker.evaluate():
+                        continue
```

Comment:
Refreshingly simple 🙂 

The `packaging` module probably tests this functionality internally, but let's add our own unit tests too (to confirm that we filter/select the correct dep based on the markers provided).

Reply:
Yeah, let's add at least one test for this.

Reply:
66f1fbd